core/src/main/java/org/jsmart/zerocode/core/utils/TokenUtils.java

```diff
@@ -34,7 +37,12 @@ public static void populateParamMap(Map<String, Object> paramaMap, String runTim
         getKnownTokens().forEach(inStoreToken -> {
                     if (runTimeToken.startsWith(inStoreToken)) {
                         if (runTimeToken.startsWith(RANDOM_NUMBER)) {
-                            paramaMap.put(runTimeToken, System.currentTimeMillis() + "");
+                        	String[] slices = runTimeToken.split(":");
+                        	if (slices.length == 2) {
+                        		paramaMap.put(runTimeToken, FixedRandomGenerator.getGenerator(Integer.parseInt(slices[1])));
+                        	}else {
+								paramaMap.put(runTimeToken, RandomGenerator.getGenerator());
```

Comment:
Basically, this calls the `toString()` of the generator?

Reply:
Yeah, `toString()` is called for every instance of the token. This functionality helps us to produce different random numbers in the same step with the same length.